src/Sleep.js

```diff
@@ -0,0 +1,103 @@
+class Sleep {
+  constructor(sleepData) {
+    this.sleepData = sleepData
+  }
+
+  calculateDataAverage(userID, dataMetric) {
+    const currentUserSleepData = [];
+    const total = this.sleepData.reduce((sum, sleepInfo) => {
+      if (sleepInfo.userID === userID) {
+        currentUserSleepData.push(sleepInfo)
+        sum += sleepInfo[dataMetric];
+      }
+      return sum;
+    }, 0);
+    return Math.round(total / currentUserSleepData.length);
+  }
+
+  filterDataByUser(userID) {
+    const currentUserSleepData = this.sleepData.filter(sleepInfo => {
+      return sleepInfo.userID === userID
+    });
+    return currentUserSleepData;
+  }
+
+  findDataByDate(date, currentUserData) {
+    const dailyData = currentUserData.find(sleepInfo => {
+      return sleepInfo.date === date
+    });
+    return dailyData;
+  }
+
+  returnDailySleepData(userID, date, dataMetric) {
+    const currentUserData = this.filterDataByUser(userID);
+    const dailySleepData = this.findDataByDate(date, currentUserData);
+    return dailySleepData[dataMetric];
+  }
+
+  findSleepDataByWeek(userID, date, dataMetric) {
+    const currentUserData = this.filterDataByUser(userID);
+    const startDate = currentUserData.find(info => info.date === date);
+    const startIndex = currentUserData.indexOf(startDate);
+    const endIndex = startIndex + 7;
+    // RETURN TO FIX ENDINDEX CALUCULATION
+    const weeklyDates = currentUserData.slice(startIndex, endIndex);
+    // CHECK WEEKLYDATES IS RETURNING EXPECTED
+    const weeklySleepData = weeklyDates.map(info => info[dataMetric])
+    return weeklySleepData;
+  }
```

Comment:
Saw this on after the branch had been merged - let's remember to come back and fix this method so we can lose the comments (lines 43, 45)!